core/controllers/pages_test.py

```diff
@@ -15,14 +15,149 @@
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+        response.mustcontain('I18N_CONTACT_PAGE_HEADING')
+
+
+class DonatePageTest(test_utils.GenericTestBase):
+    """Test for donate page."""
+
+    def test_loading_donate_page_loads_correctly(self):
+        response = self.testapp.get('/donate')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class ThanksPageTest(test_utils.GenericTestBase):
+    """Test for thanks page."""
+
+    def test_loading_thanks_page_loads_correctly(self):
+        response = self.testapp.get('/thanks')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class ForumPageTest(test_utils.GenericTestBase):
+    """Test for forum page."""
+
+    def test_loading_forum_page_loads_correctly(self):
+        response = self.testapp.get('/forum')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class TermsPageTest(test_utils.GenericTestBase):
+    """Test for terms page."""
+
+    def test_loading_terms_page_loads_correctly(self):
+        response = self.testapp.get('/terms')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+
+class PrivacyPageTest(test_utils.GenericTestBase):
+    """Test for privacy page."""
+
+    def test_loading_privacy_page_loads_correctly(self):
+        response = self.testapp.get('/privacy')
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+
+
+class AboutRedirectPageTest(test_utils.GenericTestBase):
+    """Test for about redirect page."""
+
+    def test_redirect_to_about_page_loads_correctly(self):
+        response = self.testapp.get('/credits')
+        self.assertEqual(response.status_int, 302)
+
+
+class FoundationRedirectPageTest(test_utils.GenericTestBase):
+    """Test for foundation redirect page."""
+
+    def test_redirect_to_foundation_page_loads_correctly(self):
+        response = self.testapp.get('/foundation')
+        self.assertEqual(response.status_int, 302)
+
+
+class TeachRedirectPageTest(test_utils.GenericTestBase):
+    """Test for teach redirect page."""
+
+    def test_redirect_to_teach_page_loads_correctly(self):
+        response = self.testapp.get('/participate')
+        self.assertEqual(response.status_int, 302)
+
+
+class ConsoleErrorPageTest(test_utils.GenericTestBase):
+    """Test for console error page."""
+
+    def test_loading_console_error_page_loads_correctly(self):
+        response = self.testapp.get('/console_errors')
+        self.assertEqual(response.status_int, 200)
+
+
+class MaintenancePageTest(test_utils.GenericTestBase):
+    """Test for maintenance page."""
+
+    def test_loading_maintenance_page_loads_correctly(self):
```

Comment:
Also add a test that it does not load if the constant is not set.

Reply:
when I was set the constant value false and wrote a test for it, then it not shows any error. Could explain once. Thank you

Reply:
You're swapping the constant out for True in this test. You should have another test which swaps the constant out for False, and then makes a request to the page; the page should 404.

Reply:
@netajik, have you addressed this comment? (ditto for all other open comments. do reply if they are resolved.)

Reply:
> You're swapping the constant out for True in this test. You should have another test which swaps the constant out for False, and then makes a request to the page; the page should 404.

Hello @DubeySandeep i was alreay addressed the above comment by below code. Would you review once. Thank you.


d**ef test_redirect_login_page_loads_correctly(self):
        with self.swap(feconf, 'ENABLE_MAINTENANCE_MODE', False):
            response = self.testapp.get('/admin')
            self.assertEqual(response.status_int, 302)
            self.assertIn('Login', response.headers['location'])**

Reply:
I was looking for a reply to this comment (As asked [here](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#address-review-comments-until-all-reviewers-give-lgtm-looks-good-to-me)).

Thanks for the reply!

Reply:
can you tell me once if any changes are needed?. Thank you.